src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1500,6 +1493,9 @@ void TorrentHandle::pause()
     // We test on the cached m_nativeStatus
     if (isQueued())
         m_session->handleTorrentPaused(this);
+
```

Comment:
@thalieht, can you add here the following:
```c++
if (m_pauseWhenReady) {
#if (LIBTORRENT_VERSION_NUM < 10200)
        m_nativeHandle.stop_when_ready(false);
#else
        m_nativeHandle.unset_flags(lt::torrent_flags::stop_when_ready);
#endif
        m_pauseWhenReady = false;
}
```
and check 4 cases we discussed earlier?

Reply:
1. Resumed (initial Resumed)
2. Resumed (initial Paused)
3. Paused (initial Resumed on startup)
4. Resumed (initial Paused on startup)

But
```
-  if (m_pauseWhenReady) {
+  if (!m_pauseWhenReady) {
   #if (LIBTORRENT_VERSION_NUM < 10200)
       m_nativeHandle.stop_when_ready(false);
   #else
       m_nativeHandle.unset_flags(lt::torrent_flags::stop_when_ready);
   #endif
-      m_pauseWhenReady = false;
   }
```

1. Resumed (initial Resumed)
2. Paused (initial Paused)
3. Resumed (initial Resumed on startup)
4. Paused (initial Paused on startup)

Have we decided whether it should or shouldn't resume in all cases?

Reply:
>Have we decided whether it should or shouldn't resume in all cases?

Still no.
But we can find both solutions and then just apply the chosen one.

Reply:
Can you also test the following?
```c++
#if (LIBTORRENT_VERSION_NUM < 10200)
        m_nativeHandle.stop_when_ready(false);
#else
        m_nativeHandle.unset_flags(lt::torrent_flags::stop_when_ready);
#endif
        m_pauseWhenReady = false;
```

Reply:
Resumes in all cases.

Reply:
Well, since no one is active in the survey, I choose "return to previous state" (as the path of least resistance). We can change this behavior later if there are user requests.
I rethought the solution once again: seems  you should just revert entire 87d9840 plus remove `if (m_startupState != Started) return;` from `TorrentHandle::resume()`.